Activities changed to fragments

MartaLiutyk opened the pull request:
Update Questionnaire's activities to fragments

Additional changes:
- BottomNavigationView & ViewPager has the separate fragment that is called when MainActivity starts;
- BottomNavigationView shows with text labels, with primary color when active and show response by click with Ripple animation;
- separate classes for JSON parsing with API in com.alexia.callbuttom.jsonparsers folder;
- update .gitignore;
- build.gradle up to date;
- update fragment's layouts;

Problems:
- BottomNavigationView & ViewPager transitions don't work when Questionnaire's fragment hierarchy starts

nvipash committed:
Change activities to fragments with ViewPager (no functionality)

- app works only in portrait mode
- separete folder with main fragments
- fixes with layouts
- delete unneeded layouts

Kulbach committed:
Added table locations to the database

Kulbach committed:
Added controller for locations to the API

Kulbach committed:
Fixed problems ith merge

Kulbach committed:
Updated README.md

nvipash committed:
Delete API target/ and .idea/ files

- some changes in .gitignore

MartaLiutyk committed:
Updated ButtonFragment

MartaLiutyk committed:
Added changes to manifest for work with permissions

VikaMi committed:
Updated SettingsFragment

VikaMi committed:
Merge branch 'bottomnav' of https://github.com/musiienko/Woman_Safety into bottomnav

# Conflicts:
#	API/.idea/workspace.xml

alexmusiienko committed:
added functionality for Map Fragment

nvipash committed:
Update Questionnaire's activities to fragments

Additional changes:
- BottomNavigationView & ViewPager has the separate fragment that is called when MainActivity starts;
- BottomNavigationView shows with text labels, with primary color when active and show response by click with Ripple animation;
- separate classes for JSON parsing with API in com.alexia.callbuttom.jsonparsers folder;
- update .gitignore;
- build.gradle up to date;
- update fragment's layouts;

Problems:
- BottomNavigationView & ViewPager transitions don't work when Questionnaire's fragment hierarchy starts

Kulbach committed:
Fixed bag with thansitions

Kulbach committed:
Added lost fragments to adapter

ovovchak commented on .gitignore:
```diff
@@ -33,21 +33,36 @@ proguard/
 
 # Intellij Android Studio
-*.iml
+app/app.iml
```
я думаю дописати .idea/* було б достатньо. До речі хороша практика просто гуглити щось типу ".gitignore for android project" чи там для джава проекту і просто копіювати його, щоб руцями не заповнюватию

nvipash replied:
Чомусь .idea/* не працювало як треба

nvipash replied:
UPD: поняв провтик)

ovovchak commented on API/src/main/java/safety/Controllers/LocationsController.java:
```diff
@@ -0,0 +1,53 @@
+import java.util.List;
+
+/**
+ * Created by Alexia on 19.03.2018.
```
я б не писав анотацій до класів з автором. Їх треба потім підтримувати і дописувати хто і коли той клас модифікував, а по суті нема сенсу, бо це можна в гіті глянути.

ovovchak commented on API/src/main/java/safety/Controllers/LocationsController.java:
```diff
@@ -0,0 +1,53 @@
+@RestController
+public class LocationsController {
+
+
```
у вас дуже багато порожніх лінійок в коді. 2 порожні рядочки підряд робить код дуже неакуратним. Зазвичай пишуть не більше одного рядочка

ovovchak commented on API/src/main/java/safety/Entity/LocationsEntity.java:
```diff
@@ -0,0 +1,99 @@
+
+    @Override
+    public int hashCode() {
+
```
от в такому випадку як зараз теж нема сенсу писати породнього рядочка

ovovchak commented on API/src/main/java/safety/Entity/LocationsEntity.java:
```diff
@@ -0,0 +1,99 @@
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
```
самі писали equals чи згенерили?

ovovchak commented on API/src/main/java/safety/Entity/LocationsEntity.java:
```diff
@@ -0,0 +1,99 @@
+@Entity
+@Table(name = "locations", schema = "womansafety", catalog = "")
+public class LocationsEntity {
+    private int idPlace;
```
частіше слово id пишуть в кінці

ovovchak commented on API/target/classes/hibernate.cfg.xml:
```diff
@@ -1,21 +0,0 @@
-<?xml version='1.0' encoding='utf-8'?>
```
ви взагалі конфігурашку видалили чи перенесли?

ovovchak commented on app/src/main/java/com/alexia/callbutton/MainActivity.java:
```diff
@@ -1,75 +1,94 @@
 public class MainActivity extends AppCompatActivity {
 
+    public static Bundle questionnaireResultBundle = new Bundle();
     SharedPreferences preferences;
```
перемішані всі модифікатори підряд. Java має чіткі код конвеншни. спочатку йдуть статичні речі потім паблік, потім прайвет. 

Крім того більшість змінних класу взагалі не мають чомусь модифікаторів доступу

ovovchak commented on app/src/main/java/com/alexia/callbutton/MainActivity.java:
```diff
@@ -1,75 +1,94 @@
+        bottomNav.setSelectedItemId(R.id.action_sos);
+        setUpViewPager(viewPager, bottomNav);
+        removeShiftMode(bottomNav);
+
```
ще один приклад, коли лишній рядочок порожній

ovovchak commented on app/src/main/java/com/alexia/callbutton/MainActivity.java:
```diff
@@ -132,4 +150,105 @@ public void onRequestPermissionsResult(int requestCode,
+                item.setChecked(item.getItemData().isChecked());
+            }
+        } catch (NoSuchFieldException e) {
+            Log.e("BottomNav", "Unable to get shift mode field", e);
```
насправді просто логи не мають сенсу...я б краще вже нічого з тим ексепшном не робив

ovovchak commented on app/src/main/java/com/alexia/callbutton/MainActivity.java:
```diff
@@ -132,4 +150,105 @@ public void onRequestPermissionsResult(int requestCode,
+    public static void removeShiftMode(BottomNavigationView bottomNav) {
+        BottomNavigationMenuView menuView = (BottomNavigationMenuView) bottomNav.getChildAt(0);
+        try {
+            Field shiftingMode = menuView.getClass().getDeclaredField("mShiftingMode");
```
оця штуку виглядає дуже сильно підозріло, що ви тут намагалися зробити?

nvipash replied:
Зробити, щоб у барі відображались підписи для кнопок

ovovchak commented on app/src/main/java/com/alexia/callbutton/MainActivity.java:
```diff
@@ -132,4 +150,105 @@ public void onRequestPermissionsResult(int requestCode,
+    }
+
+    @SuppressLint("ClickableViewAccessibility")
+    private void setUpViewPager(final ViewPager viewPager, final BottomNavigationView bottomNav) {
```
до речі зразу видно , що метод скопійований) у вас одразу появились final модифікатори.Загалом слово final дуже прикольна штука, його радять використовувати якомого частіше. В більш "просунутих" мовах ніж джава, взагалі більшість змінних final за замовчуванням

ovovchak commented on app/src/main/java/com/alexia/callbutton/MainActivity.java:
```diff
@@ -132,4 +150,105 @@ public void onRequestPermissionsResult(int requestCode,
+                        switch (item.getItemId()) {
+
+                            case R.id.action_help:
+                                viewPager.setCurrentItem(3, false);
```
я б якось цифри посортував...спочатку 3, потім 2, потім 0 , потім 1....хз , яка тут сакральна логіка була

ovovchak commented on app/src/main/java/com/alexia/callbutton/MainActivity.java:
```diff
@@ -132,4 +150,105 @@ public void onRequestPermissionsResult(int requestCode,
+
+            @Override
+            public void onPageSelected(int position) {
+                if(position < 4)
```
код не вирівняний і не відформатований. Або додайте {} дужки після if (що рекомендовано) або хоча б таб поставте в наступному рядочку

ovovchak commented on app/src/main/java/com/alexia/callbutton/MainActivity.java:
```diff
@@ -132,4 +150,105 @@ public void onRequestPermissionsResult(int requestCode,
+
+            }
+        });
+        viewPager.setOnTouchListener(new View.OnTouchListener() {
```
а ця штука точно тут треба? вона вроді дуже якісно нічого не робить

ovovchak commented on app/src/main/java/com/alexia/callbutton/MainActivity.java:
```diff
@@ -132,4 +150,105 @@ public void onRequestPermissionsResult(int requestCode,
+        startFragment = new QuestionnaireStartFragment();
+        surveyFragment = new QuestionnaireSurveyFragment();
+        instructionFragment = new QuestionnaireInstructionFragment();
+        adapter.addFragment(buttonFragment);
```
у вас реально в view пейджері 8 фрагментів? для чого?

nvipash replied:
то така навігація)

ovovchak commented on app/src/main/java/com/alexia/callbutton/fragments/ButtonFragment.java:
```diff
@@ -0,0 +1,76 @@
+        String toDial = "tel:" + preferences.getString("phone", "0933797479");
+        Intent callIntent = new Intent(Intent.ACTION_CALL);
+        callIntent.setData(Uri.parse(toDial));
+        if (ActivityCompat.checkSelfPermission(getActivity(), Manifest.permission.CALL_PHONE) != PackageManager.PERMISSION_GRANTED) {
```
цей if по суті якісно нічого не робить

ovovchak commented on app/src/main/java/com/alexia/callbutton/fragments/ButtonFragment.java:
```diff
@@ -0,0 +1,76 @@
+
+
+    public void call_action() {
+        Log.d("phone:", preferences.getString("phone", ""));
```
лог я б видалив після того як потестав код

ovovchak commented on app/src/main/java/com/alexia/callbutton/fragments/ButtonFragment.java:
```diff
@@ -0,0 +1,76 @@
+    }
+
+
+    public void call_action() {
```
метод названий не по джава конвеншну

ovovchak commented on app/src/main/java/com/alexia/callbutton/fragments/ButtonFragment.java:
```diff
@@ -0,0 +1,76 @@
+    }
+
+
+    public boolean isPermissionGranted() {
```
код погано поформатований, логи після тестування видаляємо. else можна забрати і написати просто після if рядочок return false;

ovovchak commented on app/src/main/java/com/alexia/callbutton/fragments/ButtonFragment.java:
```diff
@@ -0,0 +1,76 @@
+            }
+        } else { //permission is automatically granted on sdk<23 upon installation
+            Log.v("TAG", "Permission is granted");
+            return true;
```
return true не обовязково писати в else

ovovchak commented on app/src/main/java/com/alexia/callbutton/fragments/ButtonFragment.java:
```diff
@@ -0,0 +1,76 @@
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String permissions[], int[] grantResults) {
```
рядочки коду мають бути не довше 120 символів, код погано поформатований

ovovchak commented on app/src/main/java/com/alexia/callbutton/fragments/MapsFragment.java:
```diff
@@ -0,0 +1,223 @@
+import com.google.android.gms.maps.model.MarkerOptions;
+
+
+public class MapsFragment extends Fragment implements OnMapReadyCallback, GoogleApiClient.ConnectionCallbacks,
```
код погано поформатований

ovovchak commented on app/src/main/java/com/alexia/callbutton/fragments/MapsFragment.java:
```diff
@@ -0,0 +1,223 @@
+        GoogleApiClient.OnConnectionFailedListener,
+        LocationListener {
+    private GoogleMap mMap;
+    GoogleApiClient mGoogleApiClient;
```
немає модифікаторів доступу

ovovchak commented on app/src/main/java/com/alexia/callbutton/fragments/MapsFragment.java:
```diff
@@ -0,0 +1,223 @@
+    Location mLastLocation;
+    Marker mCurrLocationMarker;
+
+    public MapsFragment() {
```
непотрібний конструктор

ovovchak commented on app/src/main/java/com/alexia/callbutton/fragments/MapsFragment.java:
```diff
@@ -0,0 +1,223 @@
+        FragmentManager manager = getChildFragmentManager();
+        FragmentTransaction transaction = getChildFragmentManager().beginTransaction();
+        SupportMapFragment fragment = new SupportMapFragment();
+        transaction.add(R.id.map, fragment)
```
чому .commit з нового рядочка?

ovovchak commented on app/src/main/java/com/alexia/callbutton/fragments/MapsFragment.java:
```diff
@@ -0,0 +1,223 @@
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.maps_fragment, container, false);
+
+        FragmentManager manager = getChildFragmentManager();
```
а що таке цей SupportMapFragment і нащо він?

ovovchak commented on app/src/main/java/com/alexia/callbutton/fragments/MapsFragment.java:
```diff
@@ -0,0 +1,223 @@
+                .position(new LatLng(49.834893, 23.991080))
+                .title("Залізничний районний відділ поліції")
+                .snippet("вул. Генерала Чупринки, 65"));
+
```
лишні рядочки

ovovchak commented on app/src/main/java/com/alexia/callbutton/fragments/MapsFragment.java:
```diff
@@ -0,0 +1,223 @@
+    }
+
+    @Override
+    public void onLocationChanged(Location location) {
```
public i private речі ідуть в перемішку, мали  б бути якось посортовані

ovovchak commented on app/src/main/java/com/alexia/callbutton/fragments/QuestionnaireFragment.java:
```diff
@@ -0,0 +1,31 @@
+        selectionTest.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+               ((MainActivity) getActivity()).setCurrentPagerItem(4);
```
не до кінця розумію, що буде відбуватись...чому по натисканню кнопки ми йдемо на якийсь інший екран вюпейджера...це може і правильно написаний код, але з ux точки зору якось дивно, я б такий, що я б завів сторю в трелло, щоб це якось імпрувнути

ovovchak commented on app/src/main/java/com/alexia/callbutton/fragments/QuestionnaireInstructionFragment.java:
```diff
@@ -0,0 +1,99 @@
+import static com.google.android.gms.internal.zzagr.runOnUiThread;
+
+public class QuestionnaireInstructionFragment extends Fragment {
+    TextView surveyInstructionTitle;
```
private public? немає модифікаторів доступу. перемішані статичні і не статичні речі

ovovchak commented on app/src/main/java/com/alexia/callbutton/fragments/QuestionnaireInstructionFragment.java:
```diff
@@ -0,0 +1,99 @@
+        @Override
+        protected Void doInBackground(Void... arg0) {
+
+            HttpHandler sh = new HttpHandler();
```
чому sh?

ovovchak commented on app/src/main/java/com/alexia/callbutton/fragments/QuestionnaireSelectionFragment.java:
```diff
@@ -0,0 +1,105 @@
+    public static String url = "http://192.168.0.103:9090/api/tests/info";
+    private String TAG = QuestionnaireSelectionFragment.class.getSimpleName();
+
+    public QuestionnaireSelectionFragment() {
```
не потрібний конструктор. Крім того в файлі перемішані статичні і нестатичні речі, немає модифікаторів доступу і погано поформатований код

ovovchak commented on app/src/main/java/com/alexia/callbutton/fragments/QuestionnaireStartFragment.java:
```diff
@@ -0,0 +1,102 @@
+    public static String url = "http://192.168.0.103:9090/api/tests/info";
+    private String TAG = QuestionnaireStartFragment.class.getSimpleName();
+
+    public QuestionnaireStartFragment(){}
```
знову зауваження, що лишній конструктор, нема модифікаторів доступу, перемішані статичні і нестатичні речі

ovovchak commented on app/src/main/java/com/alexia/callbutton/fragments/QuestionnaireSurveyFragment.java:
```diff
@@ -0,0 +1,129 @@
+                runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+//                        Toast.makeText(getActivity().getApplicationContext(),
```
напевно закоментований код краще витерти, ніж комітити. потім не ясно, що з ним робити

nvipash committed:
Navigation with Back Button

Additional changes:
- code formatting based on Pull Request;
- make smaller text i BottomNavigationView;
- add navigation with back button between fragments;
- pointSum now stats from 0;

nvipash committed:
Code formatting in API

nvipash commented:
@ovovchak all fixed

nvipash committed:
Get & set questionnaire result data with Application class

Co-Authored-By: kulbach <kulbach@users.noreply.github.com>

dsvnkna committed:
add locations parsing

nvipash committed:
Replace ViewPager navigation on FragmentManager

alexmusiienko committed:
displaying markers from parsed data

nvipash committed:
Refactor SettingsFragment with PreferenceFragmentCompat

- fix bug with onBackStackPressed
- redesign and make logic of settings screen better

nvipash committed:
Some fixes with settings

Additional changes:
- update database (safe places + questionnaires)
- update UI of Questionnaire fragments (cardview & progressbar)
- fix crash when user want to add contacts first time
- add cancel button to DialogFragment with list of numbers

Problem:
- data of listview don't save after moving through the fragments

alexmusiienko committed:
added button for calling to police

nvipash committed:
Add Reference screen and add Settings button to Action Bar

Additional changes:
- preview design of "call police" button
- add action bar with title change based on viewed fragment

VikaMi committed:
add widget

Kulbach committed:
Added more answers to questions

nvipash committed:
Add splash screen when app launch

Additional changes:
- update SOS image

Kulbach committed:
Fixed bug with survey results

Kulbach committed:
Merge branch 'bottomnav' of https://github.com/musiienko/Woman_Safety into bottomnav

nvipash committed:
Implement onboarding screens

Additional changes:
- fix bug with running on < 4.4 version (Dalvik runtime)
- some fixes in layouts
- code refactoring
- change SOS image (yeah, again)
- update compile sdk version & gradle

VikaMi committed:
design widget

VikaMi committed:
Merge remote-tracking branch 'origin/bottomnav' into bottomnav

VikaMi committed:
add ic_widget_red

alexmusiienko committed:
Added sms functionality. not used anywhere yet

alexmusiienko committed:
sending message with current location

alexmusiienko committed:
Merge branch 'bottomnav' of https://github.com/musiienko/Woman_Safety into bottomnav

alexmusiienko committed:
changed text of the message

VikaMi committed:
add info to reference fragment, before test, send number to widget

dsvnkna committed:
Add images to test selection fragment

dsvnkna committed:
Merge remote-tracking branch 'origin/bottomnav' into bottomnav

# Conflicts:
#	app/src/main/res/layout/questionnaire_selection_fragment.xml
#	app/src/main/res/values/strings.xml

dsvnkna committed:
Add lost test selection fragment

VikaMi committed:
add DangerDescriptionFragment

VikaMi committed:
Merge remote-tracking branch 'origin/bottomnav' into bottomnav

nvipash committed:
Update questionnaire's fragments

Additional changes:
- update pom.xml from master branch
- code refactoring
- add new buttons for survey
- rename some classes

nvipash committed:
Fix SharedPreferences problem & design ReferenceFragment

Additional changes:
- min SDK is 19 now (can't to enable run on <19);
- add onClick ripple effects for clickable CardView;
- return back workable onItemLongClick delete for ListView;
- add hardcoded strings to string.xml;
- ActionBar hides on scroll action on ReferenceFragment & QuestionnaireViolenceInfoFragment;
- add instuction about how to add widget to the launcher to SettingsFragment ;
- ActionBar hides when questionnaire starts;
- fix size of widget_button;
- fixes with some other layouts;
- code refactoring

nvipash committed:
Merge branch 'master' into bottomnav
